components/talksPage/talksPage.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useState } from 'react';
+import style from './talksPage.module.scss';
+import data from '../../assets/videos.json';
+import YouTube from 'react-youtube';
+
+function TalksPage() {
+  const talksList = data;
+  const [currSrc, setSrc] = useState('GHbNaDSWUX8');
+
+  const calcArrowSide = (curr, side, length) => {
+    if (side > curr) {
+      if (curr === length) {
+        return 1;
+      }
+    } else if (side < curr) {
+      if (curr === 1) {
+        return length;
+      }
+    }
+    return side;
+  };
+
+  const onVideoClick = (event) => {
+    event.target.pauseVideo(); // stop chosen video
+    setSrc(event.target.playerInfo.videoData.video_id); // open the chosen video on the big iframe
+  };
+
+  const setVideos = (props) => { // add youtube videos
+    const opts = {
+      height: '200',
+      width: '320',
+    };
+
+    return props.sections.map((video, i) => (
+      <div
+        className={style.item}
+        id={'item' + i + '_' + props.index}
+        key={'item' + i + '_' + props.index}
+      >
+        <YouTube videoId={video.src} opts={opts} title={video.title} onPlay={onVideoClick}></YouTube>{' '}
+      </div>
+    ));
+  };
+
+  const setSections = (props) => { // organize sections based on categories
+    const chunkVideos = props.chunkVideos;
+    const index = props.index;
+
+    return chunkVideos.map((sections, i) => (
+      <section
+        className={style.section}
+        id={'section' + (i + 1) + '_' + index}
+        key={'section' + (i + 1) + '_' + index}
+      >
+        <a
+          href={'#section' + calcArrowSide(i + 1, i, chunkVideos.length) + '_' + index}
+          className={style.arrow__btn}
+        >
+          ‹
+        </a>
+        {setVideos({ sections, index })}
+        <a
+          href={'#section' + calcArrowSide(i + 1, i + 2, chunkVideos.length) + '_' + index}
+          className={style.arrow__btn}
+        >
+          ›
+        </a>
+      </section>
+    ));
+  };
+
+  const setCategories = (props) => { // divide each category's videos into chunks
+    const perChunk = 3; // videos per chunk
+
+    const chunkVideos = props.videos.reduce((chunkArray, video, index) => {
```

Comment:
can you explain a bit about this section?

Reply:
In each category, I devide videos into sections of 3. Every chunk has 3 videos, when you click on the right or the left arrow, you see the next chunk of 3 videos. I could do it in the Json file, but I wanted the Json to be not very complicated so that everyone could add new videos there.